my-app/src/Form.tsx

```diff
@@ -0,0 +1,150 @@
+import { Image } from './Image';
+import { Input } from './Input';
+import { Notes } from './Notes';
+import { Button } from './Button';
+import { useEffect, useState } from 'react';
+import { ImageInput } from './ImageInput';
+import { addEntry, readEntry, removeEntry, updateEntry } from './data';
+import { useNavigate, useParams } from 'react-router-dom';
+import { ModalDelete } from './ModalDelete';
+
+export function Form() {
+  const [title, setTitle] = useState<string>('');
+  const [photoUrl, setPhotoUrl] = useState(
+    'images/placeholder-image-square.jpg'
+  );
+  const [notes, setNotes] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
+  const navSubmit = useNavigate();
+
+  async function handleSave() {
+    if (id === undefined) {
+      await addEntry({ title, notes, photoUrl });
+    } else if (id !== undefined) {
```

Comment:
Developers more comfortable with the TypeScript language would not repeat the `if` clause here since it is guaranteed to be true. They would just use `if (check) {} else {}`.